src/mml_utils.py

```diff
@@ -0,0 +1,129 @@
+from bayeslite import bql_quote_name
+from bdbcontrib.population_method import population_method
+from enum import Enum
+from string import Template
+
+import json
+import jsonschema
+import pkgutil
+
+MML_SCHEMA = json.loads(
+    pkgutil.get_data('bdbcontrib', 'mml.schema.json'))
+
+
+class StatType(Enum):
+    CATEGORICAL = 1
+    NUMERICAL = 2
+    IGNORE = 3
+
+
+@population_method(population_to_bdb=0, population_name=1)
+def guess_types(bdb, table):
+    """Guesses stattypes of a given table.
+
+    Returns a dictionary from column names to the guessed stattype.
+    """
+    types = {}
+    for col in _column_names(bdb, table):
+        cursor = bdb.sql_execute(
+            'SELECT DISTINCT %s FROM %s'
+            % (bql_quote_name(col), bql_quote_name(table)))
+        # TODO(asilvers): We don't necessarily need to read all of these in,
+        # and for some datasets this may be prohibitive. But it's fine for now.
+        vals = {row[0] for row in cursor if row[0] not in [None, '']}
+        types[col] = _type_given_vals(vals)
+    return types
+
+
+def _column_names(bdb, table):
+    """Returns the column names of a table"""
+    with bdb.savepoint():
+        cursor = bdb.sql_execute('PRAGMA table_info(%s)'
+                                 % (bql_quote_name(table),))
+        return list(row[1] for row in cursor)
+
+
+def _type_given_vals(vals):
+    """Returns a guess as to the stattype of a column given its values.
+
+    These heuristics are utterly unprincipled and simply seem to work well
+    on a minimal set of test datasets. They should be treated fluidly and
+    improved as failures crop up.
+    """
```

Comment:
As I understand it, the purpose of this change is to enable users to easily manually change guessed column types.  Part of my motivation for exposing `bayesdb_guess_stattypes` from `bayeslite.guess` was to enable such a user interface in the future.

It seemed to me that the new heuristics in this change are incidental, which I guessed happened because @asilversempirical didn't know that you could just call `bayesdb_guess_stattypes`.

Otherwise I would expect to see a comment saying why that approach isn't adequate here; or, if *changing* the heuristics were an intended part of the purpose of this commit, I would expect to see those changes made to `bayeslite.guess`, which we've already been incrementally refining 'as failures crop up'.

So, based on my guess, I thought I'd point out `bayeslite.guess`, in case @asilversempirical hadn't seen it, so that we can avoid duplicating work on incrementally refined unprincipled heuristics and instead focus on more important things.

None of this should be construed to mean that I think the heuristics in `bayeslite.guess` are better (though I doubt they're particularly worse, and I suspect that more incremental refinements have gone into `bayeslite.guess` in practice), nor that I think they subsume or should subsume all manual tinkering.